images/router/haproxy/conf/haproxy-config.template

```diff
@@ -385,6 +390,11 @@ backend be_secure:{{$cfgIdx}}
     {{- end }}
   {{- end }}{{/* end disable cookies check */}}
 
+  {{- with $hsts := firstMatch $hstsPattern (index $cfg.Annotations "haproxy.router.openshift.io/hsts_header") }}
+    {{- if $hsts }}
```

Comment:
Do you need this?  https://golang.org/pkg/text/template/ says:


```
{{with pipeline}} T1 {{else}} T0 {{end}}
	If the value of the pipeline is empty, dot is unaffected and T0
	is executed; otherwise, dot is set to the value of the pipeline
	and T1 is executed.
```

Reply:
changed